contracts/loan_manager/src/contract.rs

```diff
@@ -85,24 +89,31 @@ impl LoanManager {
     }
 
     /// Upgrade deployed loan pools and the loan manager WASM.
-    pub fn upgrade(e: Env, new_manager_wasm_hash: BytesN<32>, new_pool_wasm_hash: BytesN<32>) {
-        let admin: Address = e.storage().persistent().get(&LoansDataKey::Admin).unwrap();
-        admin.require_auth();
-
-        // Upgrade the loan pools.
-        e.storage()
-            .persistent()
-            .get(&LoansDataKey::PoolAddresses)
-            .unwrap_or(vec![&e])
-            .iter()
-            .for_each(|pool| {
-                let pool_client = loan_pool::Client::new(&e, &pool);
-                pool_client.upgrade(&new_pool_wasm_hash);
-            });
-
-        // Upgrade the loan manager.
-        e.deployer()
-            .update_current_contract_wasm(new_manager_wasm_hash);
+    pub fn upgrade(
+        e: Env,
+        new_manager_wasm_hash: BytesN<32>,
+        new_pool_wasm_hash: BytesN<32>,
+    ) -> Result<(), Error> {
+        let admin: Option<Address> = e.storage().persistent().get(&LoansDataKey::Admin);
```

Comment:
If we have option values in variables like this, I think we should standardize our naming so the option-wrapper variable is named `opt_admin` and the unwrapper data is `admin`

Reply:
As it's already returning Return type I'll just switch the if let Some to '?'